src/Microsoft.AspNetCore.SignalR.Core/Internal/DefaultHubProtocolResolver.cs

```diff
@@ -33,11 +31,11 @@ public DefaultHubProtocolResolver(IOptions<HubOptions> options, IEnumerable<IHub
             }
         }
 
-        public IHubProtocol GetProtocol(string protocolName, HubConnectionContext connection)
+        public IHubProtocol GetProtocol(string protocolName, List<string> supportedProtocols, HubConnectionContext connection)
         {
             protocolName = protocolName ?? throw new ArgumentNullException(nameof(protocolName));
 
-            if (_availableProtocols.TryGetValue(protocolName, out var protocol))
+            if (_availableProtocols.TryGetValue(protocolName, out var protocol) && (supportedProtocols ==  null || supportedProtocols.Contains(protocolName)))
```

Comment:
What about casing? Do the case rules match the dictionary? It should be insensitive.